templates/servicemonitor.yaml

```diff
@@ -0,0 +1,20 @@
+{{ if .Values.serviceMonitor.enabled }}
```

Comment:
The service monitor requires a metrics endpoint to be configured and available. This should not only check whether the chart is configured to enable the service monitor, but also whether there is anything to monitor. Basically, this template could use a `.Values.configData.http.debug.prometheus.enabled` check as well.

Reply:
This was rather tricky because the complete `http.` subdict is optional and helm tends to fail with nilPointer exceptions whe you reference values whose parents aren't there. I found a workaround though i am not really happy with it:
`{{- if not ( and ( hasKey .Values "configData" ) ( hasKey .Values.configData "http" ) ( hasKey .Values.configData.http "debug" ) ( hasKey .Values.configData.http.debug "prometheus" ) ( hasKey .Values.configData.http.debug.prometheus "enabled" ) ( .Values.configData.http.debug.prometheus.enabled ) ) -}}`

Reply:
scratch that ... go and does not use lazy evaluation

Reply:
ok i blew a whole working day on this and it appears there is no way to make helm lookup a nested dict value where the parent dict does not exist... The only viable solution seems to be stacking if clauses... i will put them in tommorow after i have gone and yelled at the helm devs a little...

Reply:
I think `dig` might be what you need here. http://masterminds.github.io/sprig/dicts.html

Reply:
Holy cow... I scrolled through the libraries the whole day and missed that one completely... ok that should solve the above approach. I have gone and rewritten the PR in the meantime though, going with a more "user centric" one and allowing people to enable the debug port / metrics service endpoint without enabling the serviceMonitor at all. 

I am currently finishing it a little more and will push it then.

Reply:
ok updated the PR - this has become somewhat of a code splurge so if this feels like too much at once I am happy to break it down into smaller PRs.